src/libraries/Common/src/System/Net/Http/Http3/Helpers/VariableLengthIntegerHelper.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+
+#if KESTREL
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
+#else
+namespace System.Net.Http
+#endif
+{
+    /// <summary>
+    /// Variable length integer encoding and decoding methods. Based on https://tools.ietf.org/html/draft-ietf-quic-transport-24#section-16.
+    /// Either will take up 1, 2, 4, or 8 bytes.
+    /// </summary>
+    internal static class VariableLengthIntegerHelper
+    {
+        public const int MaximumEncodedLength = 8;
+
+        private const byte LengthMask = 0xC0;
+        private const byte LengthOneByte = 0x00;
+        private const byte LengthTwoByte = 0x40;
+        private const byte LengthFourByte = 0x80;
+        private const byte LengthEightByte = 0xC0;
+
+        private const uint TwoByteSubtract = 0x4000;
+        private const uint FourByteSubtract = 0x80000000;
+        private const ulong EightByteSubtract = 0xC000000000000000;
+
+        public const uint OneByteLimit = 64;
+        private const uint TwoByteLimit = 16383;
+        private const uint FourByteLimit = 1073741823;
+        private const long EightByteLimit = 4611686018427387903L; // 2^62-1
```

Comment:
Rather than writing out the number, could we instead write this as something like `(long)((1UL << 62) - 1)`?  Same goes for the others.  The comment at that point isn't necessary, and it's clear how the value is derived rather than having a magic value.

Reply:
In the spec it's listed as the entire integer, so it was easier to copy like that. https://tools.ietf.org/html/draft-ietf-quic-transport-24#section-16